### ISSUE TITLE
Qgis crashes when any layer is edited.

### ISSUE THREAD

**Issue by Zagalillo** [qgis/QGIS#55467]
### What is the bug or the crash?

![captura 1](https://github.com/qgis/QGIS/assets/46110763/0da63214-7a99-4e88-995d-1915115c34f0)

### Steps to reproduce the issue

Edit any layer.

### Versions

QGIS Info
QGIS Version: 3.34.0-Prizren
QGIS code revision: ffbdd67881
Compiled against Qt: 5.15.3
Running against Qt: 5.15.3
Compiled against GDAL: 3.7.2
Running against GDAL: 3.7.2

But this problem happened with older versions too.

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by PavlosDem99** [comment]
Having the same problem. When I am in editing mode of a feature and then try to choose any raster layer. I hope the video will help.

https://github.com/qgis/QGIS/assets/95135614/e58c0110-d77b-4d3f-aaee-ca69e3450da8

r, QGIS crashes.

**Comment by Zagalillo** [comment]
> What is the rule-based symbology that causes the crash?

I didn't know that the problem was caused by a rule-based sybology. I'll try to change the ones I have in the project and see what happens. 
Thanks.

**Comment by agiudiceandrea** [comment]
@PavlosDem99, please open a new issue report providing all the needed info in it.
@Zagalillo, does the issue occur also using a new QGIS user profile? If yes, please provide a sample project and layer in a zipped archive file with which the issue does occur.

**Comment by Zagalillo** [comment]
> does the issue occur also using a new QGIS user profile? If yes, please provide a sample project and layer in a zipped archive file with which the issue does occur.

Yes, I tried with a new user profile and the crash still happens. I can't provide the project because it has so many layers in several folders. I tried to zip the project using Qpackage extension but it gives me a Phyton error. 
I tried also with Qconsolidate3 extension but it gives the error "Parse error at line 1, column 1: error occurred while parsing element".
Sorry.
